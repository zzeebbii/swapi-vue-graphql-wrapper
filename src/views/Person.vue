<template>
  <div>
    <div class="error" v-if="error">{{ error }}</div>
    <div v-if="loading" class="loader"></div>
    <div class="info" v-if="person">
      <table>
        <tbody>
          <tr>
            <td>Name</td>
            <td>{{ person.name }}</td>
          </tr>
          <tr>
            <td>Height</td>
            <td>{{ person.height }}</td>
          </tr>
          <tr>
            <td>Mass</td>
            <td>{{ person.mass }}</td>
          </tr>
          <tr>
            <td>Hair Color</td>
            <td>{{ person.hair_color }}</td>
          </tr>
          <tr>
            <td>Skin Color</td>
            <td>{{ person.skin_color }}</td>
          </tr>
          <tr>
            <td>Eye Color</td>
            <td>{{ person.eye_color }}</td>
          </tr>
          <tr>
            <td>Birth Year</td>
            <td>{{ person.birth_year }}</td>
          </tr>
          <tr>
            <td>Gender</td>
            <td>{{ person.gender }}</td>
          </tr>
          <tr>
            <td>Planet</td>
            <td v-if="person.planet">{{ person.planet.name }}</td>
          </tr>
          <tr>
            <td>URL</td>
            <td>{{ person.url }}</td>
          </tr>
          <tr>
            <td>Created at</td>
            <td>{{ person.created }}</td>
          </tr>
          <tr>
            <td>Edited</td>
            <td>{{ person.edited }}</td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</template>

<script>
// @ is an alias to /src

import { SINGLE_PERSON_QUERY } from "@/utils/constants";

export default {
  name: "Person",
  data() {
    return {
      person: null,
      loading: 0,
      id: this.$route.params.id,
      error: null
    };
  },
  apollo: {
    person: {
      query: SINGLE_PERSON_QUERY,
      variables() {
        return {
          id: this.id
        };
      },
      error(error) {
        this.error = JSON.stringify(error.message);
      }
    }
  }
};
</script>
