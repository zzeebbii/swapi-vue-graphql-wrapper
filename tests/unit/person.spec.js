import { shallowMount, createLocalVue } from "@vue/test-utils";
import VueRouter from "vue-router";
import Person from "@/views/Person.vue";

const localVue = createLocalVue();
localVue.use(VueRouter);
const router = new VueRouter();

describe("Person.vue", () => {
  it("should render with data", function() {
    const PERSON_DATA = {
      name: "Test",
      height: 190,
      mass: 200,
      hair_color: "Black",
      skin_color: "white",
      eye_color: "brown",
      birth_year: "1900",
      gender: "male",
      planet: "Mars",
      url: "http://swapi.co/",
      created: "2010",
      edited: "2012"
    };

    const wrapper = shallowMount(Person, {
      localVue,
      router,
      data() {
        return {
          person: PERSON_DATA
        };
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
