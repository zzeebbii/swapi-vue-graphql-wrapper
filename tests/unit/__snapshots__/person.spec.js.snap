// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person.vue should render with data 1`] = `
<div>
  <!---->
  <!---->
  <div class="info">
    <table>
      <tbody>
        <tr>
          <td>Name</td>
          <td>Test</td>
        </tr>
        <tr>
          <td>Height</td>
          <td>190</td>
        </tr>
        <tr>
          <td>Mass</td>
          <td>200</td>
        </tr>
        <tr>
          <td>Hair Color</td>
          <td>Black</td>
        </tr>
        <tr>
          <td>Skin Color</td>
          <td>white</td>
        </tr>
        <tr>
          <td>Eye Color</td>
          <td>brown</td>
        </tr>
        <tr>
          <td>Birth Year</td>
          <td>1900</td>
        </tr>
        <tr>
          <td>Gender</td>
          <td>male</td>
        </tr>
        <tr>
          <td>Planet</td>
          <td></td>
        </tr>
        <tr>
          <td>URL</td>
          <td>http://swapi.co/</td>
        </tr>
        <tr>
          <td>Created at</td>
          <td>2010</td>
        </tr>
        <tr>
          <td>Edited</td>
          <td>2012</td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
